### PR TITLE
Wire up code UI to new runner API

### PR DESCRIPTION
**Pull request by siggisim**
It's rough v0 (no commit sha or patch), and the response is just dumped into an alert - but thought it'd be nice to have these wired up.

---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/hostedrunner/hostedrunner.go [hunk 1]
```diff
@@ -56,12 +56,19 @@ func (r *runnerService) lookupAPIKey(ctx context.Context) (string, error) {
...
-	if u.GetGroupID() == "" {
+	group := u.GetGroupID()
+	if selectedGroup != "" {
+		for _, membership := range u.GetGroupMemberships() {
+			if membership.GroupID == selectedGroup {
+				group = selectedGroup
+			}
+		}
+	}
+	if group == "" {
```
Comment (tylerwilliams) [inline enterprise/server/hostedrunner/hostedrunner.go:67]: @bduffany do we have any auth library code that will do this for us?

**Review by tylerwilliams — APPROVED** [review]
LGTM

File: enterprise/server/hostedrunner/hostedrunner.go (hunk 1 above)
Comment (siggisim) [reply enterprise/server/hostedrunner/hostedrunner.go:67]: Gunna submit this so I can play around with it, but will migrate if we have a better solution for getting a group id from either an HTTP or GRPC handler.

Comment (bduffany) [reply enterprise/server/hostedrunner/hostedrunner.go:67]: We don't have any library code to get the preferred group ID that works for both HTTP and GRPC requests, but I could have benefited from a function like that in https://github.com/buildbuddy-io/buildbuddy/pull/1183 so I may just add a library func as part of that one.

**Pull request merged**